src/AwsClient.php

```diff
@@ -564,7 +564,11 @@ private function setClientBuiltIns($args)
         $config = $this->getConfig();
         $service = $args['service'];
 
-        $builtIns['SDK::Endpoint'] = isset($args['endpoint']) ? $args['endpoint'] : null;
+        $builtIns['SDK::Endpoint'] = isset($args['endpoint'])
```

Comment:
Also, why should we not consider the endpoint resolved by the ClientResolver utility instead?

For example, the [client resolver](https://github.com/aws/aws-sdk-php/blob/b4d98bfc70df146774bf9b04f5ac5b39955fbad2/src/ClientResolver.php#L347) will ignore resolving an endpoint [if it was already provided in the arguments](https://github.com/aws/aws-sdk-php/blob/b4d98bfc70df146774bf9b04f5ac5b39955fbad2/src/ClientResolver.php#L352), otherwise it will resolve the endpoint with the default implementation [there](https://github.com/aws/aws-sdk-php/blob/b4d98bfc70df146774bf9b04f5ac5b39955fbad2/src/ClientResolver.php#L1153). 
So then the resolved config should contains the endpoint we need:
https://github.com/aws/aws-sdk-php/blob/b4d98bfc70df146774bf9b04f5ac5b39955fbad2/src/AwsClient.php#L226

What do you think?

Reply:
Re: the readability, happy to change that to an if/else, although I would rather pull the string value from the config instead of a `getEndpoint` and having to cast to a string.